src/Cache/MongoStore.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace MongoDB\Laravel\Cache;
+
+use Illuminate\Cache\DatabaseStore;
+use MongoDB\Laravel\Collection;
+use MongoDB\Laravel\Connection;
+
+use function assert;
+
+class MongoStore extends DatabaseStore
+{
+    public function __construct(
+        Connection $connection,
+        string $table,
+        string $prefix = '',
+        string $lockTable = 'cache_locks',
+        array $lockLottery = [2, 100],
+        int $defaultLockTimeoutInSeconds = 86400,
+    ) {
+        parent::__construct($connection, $table, $prefix, $lockTable, $lockLottery, $defaultLockTimeoutInSeconds);
+    }
+
+    public function put($key, $value, $seconds): bool
+    {
+        $key = $this->prefix . $key;
+        $value = $this->serialize($value);
+        $expiration = $this->getTime() + $seconds;
+        $collection = $this->table()->raw(null);
+        assert($collection instanceof Collection);
+
+        $result = $collection->updateOne(
+            ['key' => ['$eq' => $key]],
+            ['$set' => ['value' => $value, 'expiration' => $expiration]],
+            ['upsert' => true],
+        );
+
+        return $result->getUpsertedCount() > 0 || $result->getModifiedCount() > 0;
+    }
+
+    public function add($key, $value, $seconds): bool
+    {
+        $key = $this->prefix . $key;
+        $value = $this->serialize($value);
+        $expiration = $this->getTime() + $seconds;
+        $collection = $this->table()->raw(null);
+        assert($collection instanceof Collection);
+
+        $result = $collection->updateOne(
+            ['key' => ['$eq' => $key]],
+            [
+                [
+                    '$set' => [
+                        'value' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->getTime()]],
+                                'then' => $value,
+                                'else' => '$value',
+                            ],
+                        ],
+                        'expiration' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->getTime()]],
+                                'then' => $expiration,
+                                'else' => '$expiration',
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+            ['upsert' => true],
+        );
+
+        return $result->getUpsertedCount() > 0 || $result->getModifiedCount() > 0;
+    }
+
+    public function lock($name, $seconds = 0, $owner = null)
+    {
+        assert($this->connection instanceof Connection);
```

Comment:
Wouldn't this always be true since this is set through the constructor, or is it technically possible for the value to change in the parent class?

Reply:
The property is defined in the parent class. I'm removing the inheritance on `DatabaseStore`, so the code will be more optimized with less type mess.